### ISSUE TITLE
[Feature Request] Add feature on the Packager to lock the input slots

### ISSUE THREAD

**Issue by MongoTheElder** [Lothrazar/Cyclic#1023]
Automating packaging recipes that require multiple ingredients is difficult when the rate of production of the ingredients doesn't match the ratio in the recipe. For instance, Sky Orchards produces ore resources that generate a random mix of acorns and resin, generally with more acorns than resin. It takes 5 resin and 4 acorns to make one resource and the input inventory slows often gets full of acorns and block resin.

If there was a way to put the ingredients in the input inventory slots and then lock them by type the packager would no get blocked by over production of one ingredient.

Thanks for providing such an awesome mod.
MongoTheElder

**Comment by Lothrazar** [comment]
So what happens in the long term here.  I guess if you have pipes sending in resin and acorns at an equal rate, then as they get consumed acorns go slower so eventually the slots get to 100% acorns left because the resin keeps getting used.

A lock makes sense, the Hydrator has one where it fixes it stuck onto only one recipe.

So after its found a recipe and starts processing, a button shows up that lets you turn on lock mode.

And if lock mode is on, any input slots that are non empty get stuck at minimum one, and it wont scan for alternate recipes away from the one it currently has once it runs out of materials.

**Comment by Lothrazar** [comment]
Or maybe it doesnt need to lock its recipe. you could have three sets of items in there for the resin+acorn combo, so it can only make those three

**Comment by Lothrazar** [comment]
@MongoTheElder  of course the other solution is to pipe in items slower using a processing cable or other means to only add in 5/4 at once ,

**Comment by ReillyBrogan** [comment]
@Lothrazar Could we get this implemented by any chance? It would be a great QOL feature for us Sky Factory 4 players. 

It doesn't even necessarily need to be a "Lock recipe" feature, it could instead be an option to "lock" the input slots such that ghost items appear and those slots can only accept the recorded items until cleared. That way one could fill one input slot with acorns, one with resin and the others with a placeholder such as sticks.

**Comment by ReillyBrogan** [comment]
Of course the option to have the packager only craft if it has enough for 2+ of a recipe while leaving 1 recipes worth of ingredients behind would probably be the easiest option to implement.

**Comment by Lothrazar** [comment]
I fixed a bunch of performance issues with packager in another issue.  Works better and is still stable.  This is still a feature that sounds useful, will look into it for 1.14 remake. Trying to stabilize 1.12 without adding new content.   Working on brand new mods and 1.14+ ports
